### DISCUSSION TITLE
Obsoleted modules

### DISCUSSION THREAD

**Discussion by kpodsiad** [scalacenter/bloop#1644] [General] [answered]
Are modules such as `jsonConfig210` and `sbtBloop013` needed? Metals don't support earlier versions of Scala than 2.12

**Comment by tgodzik** [comment, marked as answer]
`sbtBloop013` is need to support sbt 0.13, which is probably also why sbtBloop013 is needed. Maybe it's time we dropped 0.13 ?

**Reply by kpodsiad** [reply]
> Maybe it's time we dropped 0.13 ?

That's what I pretty much wanted to ask.

**Reply by dos65** [reply]
Is there an option to check if these modules aren't used by sonatype statistic?

**Reply by tgodzik** [reply]
I don't have access to it unfortunately. But people can always use an older version of the plugin, we haven't changed the configuration in a while. It should be safe to remove it.

**Reply by dos65** [reply]
hmm. maybe we could ask someone from sbt-team if they have statistic is there are significant number in 0.13 downloads or not?
I know some users who are still use sbt 0.13 and they don't have chances to upgrade.

Also, is there any issue in supporting these modules?

**Reply by kpodsiad** [reply]
As Tomek said, people can always use an older version of the plugin, configuration hasn't changed in a while. 
Although there are no issues in these modules, I think we should aim to reduce complexity in bloop.

**Reply by dos65** [reply]
If there is no issue then let's keep them. I don't think that dropping two modules that have few files changes the complexity.

> people can always use an older version of the plugin

This requires advanced knowledge of Metals and it will take a time to found out how and which version users should set.

**Reply by kpodsiad** [reply]
There are no problems with modules alone, but every module contributes to the overall complexity - build config, CI actions, release tasks. 
I understand your concerns but I'm just afraid that if we don't start cutting off deadwood we'll never be able to iterate faster. 

However, it's not necessary and we can wait with removing them.

**Reply by tgodzik** [reply]
> This requires advanced knowledge of Metals and it will take a time to found out how and which version users should set.

Actually it doesn't, we already hardcode the sbt-bloop version for sbt versions 1.0.0- 1.3.0, so we could do the same. If anyone adds the sbt-bloop version manually, we can mention it in the docs for them.

> If there is no issue then let's keep them. I don't think that dropping two modules that have few files changes the complexity.

We can't also keep it forever, it's better to try and force users of the older versions (especially 0.13.x, which I haven't seen used in a bit) and besides Metals will correctly work with it still.

**Reply by dos65** [reply]
> Actually it doesn't, we already hardcode the sbt-bloop version for sbt versions 1.0.0- 1.3.0, so we could do the same.

That is an option. Let's do that first in case if we are going to drop these modules in the next release.

**Reply by dos65** [reply]
Oops. The whole discussion was pointless because we already use `1.4.6` bloop plugin for sbt which version is less than `1.3.0` - https://github.com/scalameta/metals/blob/main/metals/src/main/scala/scala/meta/internal/builds/SbtBuildTool.scala#L158

Sorry!
